libwasmvm/src/gas_report.rs

```diff
@@ -0,0 +1,24 @@
+#[derive(Copy, Clone, Debug)]
+#[repr(C)]
+pub struct GasReport {
+    /// The original limit the instance was created with
+    pub limit: u64,
+    /// The remaining gas that can be spend
+    pub remaining: u64,
+    /// The amount of gas that was spend and metered externally in operations triggered by this instance
+    pub used_externally: u64,
+    /// The amount of gas that was spend and metered internally (i.e. by executing Wasm and calling
+    /// API methods which are not metered externally)
+    pub used_internally: u64,
+}
+
+impl From<cosmwasm_vm::GasReport> for GasReport {
+    fn from(value: cosmwasm_vm::GasReport) -> Self {
```

Comment:
In `impl From<cosmwasm_vm::AnalysisReport> for AnalysisReport` and `impl From<cosmwasm_vm::Metrics> for Metrics` I decomposed the input first to ensure all fields are used. Not sure if that helps or is needed though.

Reply:
That's a very good point, I didn't think about that.